### PR TITLE
better error reporting for ai exceptions

### PR DESCRIPTION
**Pull request by marvin-bitterlich**
This needs to be on the server now @Raycoms

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 1]
```diff
@@ -94,6 +94,11 @@
     private WalkToProxy proxy;
 
     /**
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:96]: MINOR First sentence should end with a period. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 2]
```diff
@@ -150,8 +155,16 @@ protected AbstractEntityAIBasic(@NotNull final J job)
...
+        final int timeout = EXCEPTION_TIMEOUT * exceptionTimer;
+        // wait for longer now
+        exceptionTimer *= 2;
+        final String name = this.worker.getName();
+        final BlockPos workerPosition = worker.getPosition();
+        final String jobName = worker.getColonyJob().getName();
+        Log.getLogger().info("Pausing Entity " + name + " (" + jobName + ") at " + workerPosition + " for " + timeout + " Seconds because of error:");
+        this.setDelay(timeout);
+        // fix for printing the actual exception
+        e.printStackTrace();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: CRITICAL Use a logger to log this exception. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1148)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java (hunk 2 above)
Comment (marvin-bitterlich) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: no I wont xD

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 3]
```diff
@@ -150,8 +155,33 @@ protected AbstractEntityAIBasic(@NotNull final J job)
...
+                final String jobName = colonyJob == null ? "null" : colonyJob.getName();
+                Log.getLogger().error("Pausing Entity " + name + " (" + jobName + ") at " + workerPosition + " for " + timeout + " Seconds because of error:");
+            }
+            else
+            {
+                Log.getLogger().error("Pausing Entity that is null for " + timeout + " Seconds because of error:");
+            }
+
+            // fix for printing the actual exception
+            e.printStackTrace();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:178]: CRITICAL Use a logger to log this exception. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1148)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 4]
```diff
@@ -150,8 +155,33 @@ protected AbstractEntityAIBasic(@NotNull final J job)
...
+        }catch (RuntimeException exp){
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:179]: MINOR Move this left curly brace to the beginning of next line of code. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)
MINOR Move this "catch" keyword to a new dedicated line. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ARightCurlyBraceDifferentLineAsNextBlockCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 5]
```diff
@@ -150,8 +155,33 @@ protected AbstractEntityAIBasic(@NotNull final J job)
...
     {
-        Log.getLogger().info("Pausing Entity for 5 Seconds");
-        this.setDelay(EXCEPTION_TIMEOUT);
+        try
+        {
+            final int timeout = EXCEPTION_TIMEOUT * exceptionTimer;
+            this.setDelay(timeout);
+            // wait for longer now
+            exceptionTimer *= 2;
+            if (worker != null)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:164]: MAJOR Avoid if (x != y) ..; else ..; [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AConfusingTernary)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 6]
```diff
@@ -150,8 +155,33 @@ protected AbstractEntityAIBasic(@NotNull final J job)
...
+            else
+            {
+                Log.getLogger().error("Pausing Entity that is null for " + timeout + " Seconds because of error:");
+            }
+
+            // fix for printing the actual exception
+            e.printStackTrace();
+        }catch (RuntimeException exp){
+            Log.getLogger().error("Welp reporting crashed:");
+            exp.printStackTrace();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:181]: CRITICAL Use a logger to log this exception. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1148)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 7]
```diff
@@ -150,8 +155,33 @@ protected AbstractEntityAIBasic(@NotNull final J job)
...
+            Log.getLogger().error("Caused by ai exception:");
+            e.printStackTrace();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:183]: CRITICAL Use a logger to log this exception. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1148)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 7 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request merged**